src/bindings/python/src/pyopenvino/graph/model.cpp

```diff
@@ -327,19 +327,51 @@ void regclass_graph_Model(py::module m) {
     model.def(
         "reshape",
         [](ov::Model& self, const py::list& partial_shape) {
-            ov::PartialShape new_shape(Common::partial_shape_from_list(partial_shape));
-            py::gil_scoped_release release;
-            self.reshape(new_shape);
+            if (py::len(partial_shape) > 0 && (py::isinstance<py::list>(partial_shape[0]) || py::isinstance<ov::PartialShape>(partial_shape[0]) || py::isinstance<py::str>(partial_shape[0]) || py::isinstance<py::tuple>(partial_shape[0]))) {
```

Comment:
> py::isinstance<ov::PartialShape>(partial_shape[0])

It looks excessive as originally it was list of PartialShapes and now looks like else-statement won't be reached.